lib/SILOptimizer/Mandatory/TFCanonicalizeCFG.cpp

```diff
@@ -189,6 +198,18 @@ namespace {
                                      SILBasicBlock *endBB);
     void processLoop(SILLoop *loop);
     void ensureSingleExitFromLoops();
+
+    // Dump top-level loop information for debugging purposes.
+    void dumpTopLevelLoopInfo(llvm::raw_ostream* outs, const char* stage) {
```

Comment:
would it be useful to have a routine that dumps loop info for a specific loop? this way after transforming one loop, we can just dump out it's structure before and after.

Reply:
We can already do that with `loop->dump() or loop->print(...)`?

Reply:
would it be useful for debugging, if we dump the loop info out before and after the transformation, for each transformation step?